var/spack/repos/builtin/packages/opencv/package.py

```diff
@@ -2,399 +2,1105 @@
-    version('2.4.12.2', sha256='150a165eb14a5ea74fb94dcc16ac7d668a6ff20a4449df2570734a2abaab9c0e', deprecated=True)
-    version('2.4.12.1', sha256='c1564771f79304a2597ae4f74f44032021e3a46657e4a117060c08f5ed05ad83', deprecated=True)
+    homepage = "https://opencv.org/"
+    url = "https://github.com/opencv/opencv/archive/4.5.0.tar.gz"
+    git = "https://github.com/opencv/opencv.git"
+
+    maintainers = ["bvanessen", "adamjstewart"]
+
+    version("master", branch="master")
+    version(
+        "4.5.4",
+        sha256="c20bb83dd790fc69df9f105477e24267706715a9d3c705ca1e7f613c7b3bad3d",
+    )
+    version(
+        "4.5.2",
+        sha256="ae258ed50aa039279c3d36afdea5c6ecf762515836b27871a8957c610d0424f8",
+    )
+    version(
+        "4.5.1",
+        sha256="e27fe5b168918ab60d58d7ace2bd82dd14a4d0bd1d3ae182952c2113f5637513",
+    )
+    version(
+        "4.5.0",
+        sha256="dde4bf8d6639a5d3fe34d5515eab4a15669ded609a1d622350c7ff20dace1907",
+    )
+    version(
+        "4.2.0",
+        sha256="9ccb2192d7e8c03c58fee07051364d94ed7599363f3b0dce1c5e6cc11c1bb0ec",
+    )
+    version(
+        "4.1.2",
+        sha256="385dd0a9c25e67ef0dd60e022d2a2d7b17e2f36819cf3cb46aa8cdff5c5282c9",
+    )
+    version(
+        "4.1.1",
+        sha256="5de5d96bdfb9dad6e6061d70f47a0a91cee96bb35afb9afb9ecb3d43e243d217",
+    )
+    version(
+        "4.1.0",
+        sha256="8f6e4ab393d81d72caae6e78bd0fd6956117ec9f006fba55fcdb88caf62989b7",
+    )
+    version(
+        "4.0.1",
+        sha256="7b86a0ee804244e0c407321f895b15e4a7162e9c5c0d2efc85f1cadec4011af4",
+    )
+    version(
+        "4.0.0",
+        sha256="3787b3cc7b21bba1441819cb00c636911a846c0392ddf6211d398040a1e4886c",
+    )
+    version(
+        "3.4.12",
+        sha256="c8919dfb5ead6be67534bf794cb0925534311f1cd5c6680f8164ad1813c88d13",
+    )
+    version(
+        "3.4.6",
+        sha256="e7d311ff97f376b8ee85112e2b536dbf4bdf1233673500175ed7cf21a0089f6d",
+    )
+    version(
+        "3.4.5",
+        sha256="0c57d9dd6d30cbffe68a09b03f4bebe773ee44dc8ff5cd6eaeb7f4d5ef3b428e",
+    )
+    version(
+        "3.4.4",
+        sha256="a35b00a71d77b484f73ec485c65fe56c7a6fa48acd5ce55c197aef2e13c78746",
+    )
+    version(
+        "3.4.3",
+        sha256="4eef85759d5450b183459ff216b4c0fa43e87a4f6aa92c8af649f89336f002ec",
+    )
+    version(
+        "3.4.1",
+        sha256="f1b87684d75496a1054405ae3ee0b6573acaf3dad39eaf4f1d66fdd7e03dc852",
+    )
+    version(
+        "3.4.0",
+        sha256="678cc3d2d1b3464b512b084a8cca1fad7de207c7abdf2caa1fed636c13e916da",
+    )
+    version(
+        "3.3.1",
+        sha256="5dca3bb0d661af311e25a72b04a7e4c22c47c1aa86eb73e70063cd378a2aa6ee",
+    )
+    version(
+        "3.3.0",
+        sha256="8bb312b9d9fd17336dc1f8b3ac82f021ca50e2034afc866098866176d985adc6",
+    )
+    version(
+        "3.2.0",
+        sha256="9541efbf68f298f45914b4e837490647f4d5e472b4c0c04414a787d116a702b2",
+        deprecated=True,
```

Comment:
You can actually remove these deprecated versions now if you want

Reply:
Done.